sdk/trace/trace_test.go

```diff
@@ -135,6 +136,42 @@ func TestRecordingIsOff(t *testing.T) {
 }
 
 // TODO: [rghetia] enable sampling test when Sampling is working.
+func TestSampling(t *testing.T) {
+	total := 10000
+	for name, tc := range map[string]struct {
+		sampler   Sampler
+		expect    float64
+		tolerance float64
+	}{
+		"AlwaysSample":           {sampler: AlwaysSample(), expect: 1.0, tolerance: 0},
+		"NeverSample":            {sampler: NeverSample(), expect: 0, tolerance: 0},
+		"ProbabilitySampler(25)": {sampler: ProbabilitySampler(0.25), expect: .25, tolerance: 0.015},
+		"ProbabilitySampler(50)": {sampler: ProbabilitySampler(0.50), expect: .5, tolerance: 0.015},
+		"ProbabilitySampler(75)": {sampler: ProbabilitySampler(0.75), expect: .75, tolerance: 0.015},
```

Comment:
Please add test cases for probability value 1 and 0. Probability value of >1 and <0 doesn't make sense but the api accepts it. So either have api reject it or add those to test cases as well.

Also add test case where the parent is Sampled and the Sampler is configured with probability of 0 and >0.

Reply:
I believe I have addressed all of those.